akd/src/append_only_zks.rs

```diff
@@ -289,6 +291,26 @@ impl Azks {
         Ok(())
     }
 
+    pub(crate) async fn preload_lookup_nodes<S: Database + Send + Sync>(
+        &self,
+        storage: &StorageManager<S>,
+        lookup_infos: &[LookupInfo],
+    ) -> Result<u64, AkdError> {
+        // Collect lookup labels needed and convert them into Nodes for preloading.
+        let lookup_nodes: Vec<Node> = lookup_infos
+            .iter()
+            .flat_map(|li| vec![li.existent_label, li.marker_label, li.non_existent_label])
+            .map(|l| Node {
+                label: l,
+                hash: EMPTY_DIGEST,
+            })
+            .collect();
+
+        // Load nodes. Note NodeSet will sort these nodes for efficient preloading.
```

Comment:
I think we can drop the sorting comment, how `NodeSet` represents the data is an internal detail.